examples/nlp/language_modeling/megatron_ckpt_to_nemo.py

```diff
@@ -82,91 +87,100 @@ def get_args():
-
     app_state.model_parallel_size = app_state.tensor_model_parallel_size * app_state.pipeline_model_parallel_size
+    app_state.world_size = args.tensor_model_parallel_size * args.pipeline_model_parallel_size
+    app_state.tensor_model_parallel_rank = 0
+    app_state.pipeline_model_parallel_rank = 0
 
-    parallel_state.initialize_model_parallel(
-        tensor_model_parallel_size_=app_state.tensor_model_parallel_size,
-        pipeline_model_parallel_size_=app_state.pipeline_model_parallel_size,
-        pipeline_model_parallel_split_rank_=app_state.pipeline_model_parallel_split_rank,
-    )
-
-    app_state.pipeline_model_parallel_rank = parallel_state.get_pipeline_model_parallel_rank()
-    app_state.tensor_model_parallel_rank = parallel_state.get_tensor_model_parallel_rank()
+    save_connector = NLPSaveRestoreConnector()
+    trainer = Trainer(devices=1, strategy=NLPDDPStrategy(), accelerator="cpu")
 
-    # inject model parallel rank
     checkpoint_path = inject_model_parallel_rank(os.path.join(args.checkpoint_folder, args.checkpoint_name))
-
-    logging.info(
-        f'rank: {rank}, local_rank: {local_rank}, is loading checkpoint: {checkpoint_path} for tp_rank: {app_state.tensor_model_parallel_rank} and pp_rank: {app_state.pipeline_model_parallel_rank}'
-    )
-
     if args.model_type == 'gpt':
-        model = MegatronGPTModel.load_from_checkpoint(checkpoint_path, hparams_file=args.hparams_file, trainer=trainer)
+        model = MegatronGPTModel.load_from_checkpoint(
+            checkpoint_path, hparams_file=args.hparams_file, trainer=trainer
+        ).to("cpu")
     elif args.model_type == 'bert':
         model = MegatronBertModel.load_from_checkpoint(
             checkpoint_path, hparams_file=args.hparams_file, trainer=trainer
-        )
+        ).to("cpu")
     elif args.model_type == 't5':
-        model = MegatronT5Model.load_from_checkpoint(checkpoint_path, hparams_file=args.hparams_file, trainer=trainer)
+        model = MegatronT5Model.load_from_checkpoint(
+            checkpoint_path, hparams_file=args.hparams_file, trainer=trainer
+        ).to("cpu")
     elif args.model_type == 'bart':
         model = MegatronBARTModel.load_from_checkpoint(
             checkpoint_path, hparams_file=args.hparams_file, trainer=trainer
-        )
+        ).to("cpu")
     elif args.model_type == 'nmt':
-        model = MegatronNMTModel.load_from_checkpoint(checkpoint_path, hparams_file=args.hparams_file, trainer=trainer)
+        model = MegatronNMTModel.load_from_checkpoint(
+            checkpoint_path, hparams_file=args.hparams_file, trainer=trainer
+        ).to("cpu")
     elif args.model_type == 'retro':
         model = MegatronRetrievalModel.load_from_checkpoint(
             checkpoint_path, hparams_file=args.hparams_file, trainer=trainer
-        )
-    model._save_restore_connector = NLPSaveRestoreConnector()
+        ).to("cpu")
+
+    checkpoint_dir = args.checkpoint_folder
```

Comment:
Is the following section necessary or can be replaced with just Nemo model.save_to("basepth") and let it automatically calculate tp pp path + save the checkpoint and then finally repackage? 

If that doesn't work, below code is fine

Reply:
I think the only thing keep us from using save_to is the weird pathing https://github.com/NVIDIA/NeMo/blob/b9cf05cf76496b57867d39308028c60fef7cb1ba/nemo/collections/nlp/parts/nlp_overrides.py#L245.  I do the same thing as here but change the paths properly